smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/InotifyEventFetcher.java

```diff
@@ -203,13 +228,95 @@ public void run() {
     private final InotifyEventApplier applier;
     private final QueueFile queueFile;
     private long lastId;
+    private SmartConf conf;
+    private List<String> ignoreList;
 
     public EventApplyTask(NamespaceFetcher namespaceFetcher, InotifyEventApplier applier,
         QueueFile queueFile, long lastId) {
       this.namespaceFetcher = namespaceFetcher;
       this.queueFile = queueFile;
       this.applier = applier;
       this.lastId = lastId;
+      this.conf = new SmartConf();
+      this.ignoreList = getIgnoreDirFromConfig();
+    }
+
+    public EventApplyTask(NamespaceFetcher namespaceFetcher, InotifyEventApplier applier,
+        QueueFile queueFile, long lastId, SmartConf conf) {
+      this.namespaceFetcher = namespaceFetcher;
+      this.queueFile = queueFile;
+      this.applier = applier;
+      this.lastId = lastId;
+      this.conf = conf;
+      this.ignoreList = getIgnoreDirFromConfig();
+    }
+
+    public List<String> getIgnoreDirFromConfig() {
+      String ignoreDirs = this.conf.get(SmartConfKeys.SMART_IGNORE_DIRS_KEY);
+      List<String> ignoreList;
+      if (ignoreDirs == null || ignoreDirs.equals("")) {
+        ignoreList = new ArrayList<>();
+      } else {
+        ignoreList = Arrays.asList(ignoreDirs.split(","));
+      }
+      return ignoreList;
+    }
+
+    public boolean fetchPathInIgnoreList(String path) {
+      if (!path.endsWith("/")) {
+        path = path.concat("/");
+      }
+      for (int i = 0; i < ignoreList.size(); i++) {
+        String ignoreDir = ignoreList.get(i);
+        if (!ignoreDir.endsWith("/")) {
+          ignoreDir = ignoreDir.concat("/");
+        }
+        if (path.equals(ignoreDir)) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    public boolean ifEventIgnore(Event event) {
+      String path;
+      switch (event.getEventType()) {
+        case CREATE:
+          Event.CreateEvent createEvent = (Event.CreateEvent) event;
+          path = createEvent.getPath();
+          return fetchPathInIgnoreList(path);
+        case CLOSE:
+          Event.CloseEvent closeEvent = (Event.CloseEvent) event;
+          path = closeEvent.getPath();
+          return fetchPathInIgnoreList(path);
+        case RENAME:
```

Comment:
For rename, the event cannot be ignored.

Reply:
Actually, the event can be ignored if the source and dest path are all in the directories that should be ignored.